doc/source/tutorial/stats/sampling_hinv.rst

```diff
@@ -0,0 +1,191 @@
+``|U - CDF(X)|`` where ``X`` is the approximate percentile corressponding to the
+quantile ``U`` i.e. ``X = approx_ppf(U)``) can be set to the
+required resolution (within machine precision). Notice that very small values of
+the u-resolution are possible but may increase the cost for the setup step.
+
+`NumericalInverseHermite` approximates the inverse of a continuous
+statistical distribution's CDF with a Hermite spline. Order of the
+hermite spline can be specified by passing the ``order`` parameter.
+
+As described in [1]_, it begins by evaluating the distribution's PDF and
+CDF at a mesh of quantiles ``x`` within the distribution's support.
+It uses the results to fit a Hermite spline ``H`` such that
+``H(p) == x``, where ``p`` is the array of percentiles corresponding
+with the quantiles ``x``. Therefore, the spline approximates the inverse
+of the distribution's CDF to machine precision at the percentiles ``p``,
+but typically, the spline will not be as accurate at the midpoints between
+the percentile points::
+
+    p_mid = (p[:-1] + p[1:])/2
+
+so the mesh of quantiles is refined as needed to reduce the maximum
+"u-error"::
+
+    u_error = np.max(np.abs(dist.cdf(H(p_mid)) - p_mid))
+
+below the specified tolerance `u_resolution`. Refinement stops when the required
+tolerance is achieved or when the number of mesh intervals after the next
+refinement could exceed the maximum allowed number `max_intervals`.
+
+    >>> from scipy.stats import NumericalInverseHermite
+    >>> from scipy.stats import norm, genexpon
+    >>> from scipy.special import ndtr
+
+To create a generator to sample from the standard normal distribution, do:
+
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...        return 1/np.sqrt(2*np.pi) * np.exp(-x**2 / 2)
+    ...     def cdf(self, x):
+    ...        return ndtr(x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInverseHermite(dist, random_state=urng)
+
+The `NumericalInverseHermite` has a method that approximates the PPF of the
+distribution.
+
+    >>> rng = NumericalInverseHermite(dist)
+    >>> p = np.linspace(0.01, 0.99, 99) # percentiles from 1% to 99%
+    >>> np.allclose(rng.ppf(p), norm.ppf(p))
+    True
+
+Depending on the implementation of the distribution's random sampling
+method, the random variates generated may be nearly identical, given
+the same random state.
+
+    >>> dist = genexpon(9, 16, 3)
+    >>> rng = NumericalInverseHermite(dist)
+    >>> # `seed` ensures identical random streams are used by each `rvs` method
+    >>> seed = 500072020
+    >>> rvs1 = dist.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> rvs2 = rng.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+To check that the random variates closely follow the given distribution, we can
+look at its histogram:
+
+.. plot::
+
+    >>> import numpy as np
```

Comment:
I need to reimport everything and rewrite the code from scratch again when using the `.. plot::` directive. One way to avoid this could be to put all the code below the `.. plot::` directive in a file under `doc/source/tutorial/stats/plots` and include the resulting plot in the tutorial without showing the code:

    .. plot:: tutorial/stats/plots/hinv_plot.py
    :align: center
    :include-source: 0

Reply:
Didn't know this yet! Thanks.

Reply:
I have changed it this way. Let's see how it looks.